src/ecpool.erl

```diff
@@ -93,33 +80,39 @@ add_reconnect_callback(Pool, Callback) ->
      || {_WorkerName, Worker} <- ecpool:workers(Pool)],
     ok.
 
+%% NOTE: Use pick_and_do/3 instead of with_client/2,3
+%%   to avoid applying action failure with 'badfun'.
```

Comment:
I have not read the cod in details, just checking:
do we save the passed in Action (of `fun()` spec)?

Reply:
It won't. But you reminded me that the on_reconnect/on_disconnect callbacks, which are saved in the state of ecpool worker, should be fixed.